gunrock/app/hits/hits_enactor.cuh

```diff
@@ -138,52 +141,113 @@ struct hitsIterationLoop
+
+        auto normalize_divide_square_op =
+        [hrank_next, arank_next, hrank_mag, arank_mag] __host__ __device__(
+            VertexT * v_q, const SizeT &pos) {
+          if (hrank_mag[0] > 0) {
+            hrank_next[pos] = sqrt(hrank_next[pos]) / sqrt(hrank_mag[0]);
+          }
+
+          if (arank_mag[0] > 0) {
+            arank_next[pos] = sqrt(arank_next[pos]) / sqrt(arank_mag[0]);
+          }
+        };
+        // Divide all elements by the square root of their squared sums.
+        // Note: take sqrt of x in denominator because x^2 was done in place.
+        GUARD_CU(frontier.V_Q()->ForAll(normalize_divide_square_op, graph.nodes));
+
+        GUARD_CU2(cudaStreamSynchronize(stream), "cudaStreamSynchronize Failed");
+      }
+      else if(hits_norm == 1) {
+        // Sum all scores in each array
+        GUARD_CU(util::cubReduce(
+          cub_temp_space, hrank_next, hrank_mag, graph.nodes,
           [] __host__ __device__(const ValueT &a, const ValueT &b) {
-            return a + b;
+            return abs(a) + abs(b);
           },
           ValueT(0), stream));
 
-      GUARD_CU2(cudaStreamSynchronize(stream), "cudaStreamSynchronize Failed");
+        GUARD_CU(util::cubReduce(
+            cub_temp_space, arank_next, arank_mag, graph.nodes,
+            [] __host__ __device__(const ValueT &a, const ValueT &b) {
+              return abs(a) + abs(b);
+            },
+            ValueT(0), stream));
+
+        GUARD_CU2(cudaStreamSynchronize(stream), "cudaStreamSynchronize Failed");
+
+        auto normalize_divide_op =
+        [hrank_next, arank_next, hrank_mag, arank_mag] __host__ __device__(
+            VertexT * v_q, const SizeT &pos) {
+          if (hrank_mag[0] > 0) {
+            hrank_next[pos] = hrank_next[pos] / hrank_mag[0];
+          }
 
-      auto normalize_divide_op =
-          [hrank_next, arank_next, hrank_mag, arank_mag] __host__ __device__(
-              VertexT * v_q, const SizeT &pos) {
-            if (hrank_mag[0] > 0) {
-              hrank_next[pos] = sqrt(hrank_next[pos]) / sqrt(hrank_mag[0]);
-            }
-
-            if (arank_mag[0] > 0) {
-              arank_next[pos] = sqrt(arank_next[pos]) / sqrt(arank_mag[0]);
-            }
-          };
-      // Divide all elements by the square root of their squared sums.
-      // Note: take sqrt of x in denominator because x^2 was done in place.
-      GUARD_CU(frontier.V_Q()->ForAll(normalize_divide_op, graph.nodes));
+          if (arank_mag[0] > 0) {
+            arank_next[pos] = arank_next[pos] / arank_mag[0];
+          }
+        };
+        // Divide all elements by the square root of their squared sums.
+        // Note: take sqrt of x in denominator because x^2 was done in place.
+        GUARD_CU(frontier.V_Q()->ForAll(normalize_divide_op, graph.nodes));
+
+        GUARD_CU2(cudaStreamSynchronize(stream), "cudaStreamSynchronize Failed");
+      }
+      else {
+        assert(false); // TODO: Handle errors case
```

Comment:
Is this when `hits_norm != 1,2`?

Reply:
Yes. Is there a more graceful convention for how Gunrock handles unsupported cases?